test/run-tests.py

```diff
@@ -615,28 +675,66 @@ def main(args):
     out_dir = tempfile.mkdtemp(prefix='sexpr-wasm-')
     out_dir_is_temp = True
 
+  variables['out_dir'] = os.path.abspath(out_dir)
+
+  allProcs = []
+  continuedErrors = 0
   try:
-    for i, p in enumerate(range(num_proc)):
-      args = (i, options, variables, inq, outq, out_dir)
-      proc = multiprocessing.Process(target=ProcessWorker, args=args)
-      proc.start()
+    if not options.no_multithreaded:
+      inq = multiprocessing.Queue()
+      outq = multiprocessing.Queue()
+      for info in infosToRun:
+        inq.put_nowait(info)
+      runWorkers = multiprocessing.Value('i', 1)
```

Comment:
I think running is misleading, this doesn't tell if the worker are running, but tells if the workers should run.
How about `do_work` ?